JitsiConnection.js

```diff
@@ -25,12 +26,19 @@ function JitsiConnection(appID, token, options) {
 
     this.addEventListener(JitsiConnectionEvents.CONNECTION_FAILED,
         function (errType, msg) {
+            AnalyticsAdapter.sendEvent('connection.failed.' + errType);
             if(errType === JitsiConnectionErrors.OTHER_ERROR &&
                 (msg === "item-not-found" || msg === "host-unknown")) {
                     // FIXME: don't report the error if we are going to reload
                     this._reload();
                 }
         }.bind(this));
+
+    this.addEventListener(JitsiConnectionEvents.CONNECTION_DISCONNECTED,
+        function (msg) {
+            if(msg)
+                AnalyticsAdapter.sendEvent('connection.disconnected.' + msg);
```

Comment:
Nope, the case is we eventually can have an error message. And there are cases when you just close the tab and strophe had enough time to close and if there is no check for message you see in the stats event: 'connection.disconnected.undefined'